packages/material-ui/src/BottomNavigationAction/BottomNavigationAction.d.ts

```diff
@@ -7,12 +7,32 @@ export type BottomNavigationActionTypeMap<
   D extends React.ElementType
 > = ExtendButtonBaseTypeMap<{
   props: P & {
-    icon?: string | React.ReactElement;
+    // TODO: never
+    /**
+     * This prop isn't supported.
+     * Use the `component` prop if you need to change the children structure.
+     */
+    children?: React.ReactNode;
+    /**
+     * The icon element.
+     */
+    icon?: React.ReactNode;
```

Comment:
To sync with propTypes. I don't see a reason to disallow arrays or booleans from conditional rendering.